### PR TITLE
fix: resolve `jsxImportSource` relative to module

### LINKED ISSUE #15559: `jsxImportSource` resolution in config files
Take the following example:

```json
// deno.json
{
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "preact"
  },
  "importMap": "./import_map.json"
}
```

```json
// import_map.json
{
  "scopes": {
    "./x/": {
      "preact": "./preact-1.js"
    },
    "./y/": {
      "preact": "./preact-2.js"
    }
  }
}
```

Currently in Deno such code would fail because the bare specifier "preact" can not be resolved from the config file module.

This is arguably a very rarely encountered edge case. I still think we should deal with it in some way. I can think of two ways:

1. Resolve `jsxImportSource` relative to the config file location, and use the resolved specifier during emit. This poses the risk that the specifier may be resolved by the import map twice: once during emit, and once at runtime when the jsx import is actually loaded.
2. Do not resolve `jsxImportSource` up front at all. Emit the specifier exactly as specified in the config file into the source files. Resolve the specifier individually in every file, just like `/** @jsxImportSource ... */` is handled.

I prefer option 2, because it has less weird edge cases. Both may have surprising behavior when using a relative specifier in the config file. I think that is the least of all evils though.

### PR DESCRIPTION
**Pull request by lucacasonato**
Previously `jsxImportSource` was resolved relative to the config file
during graph building, and relative to the emitted module during
runtime.

This is now fixed so that the JSX import source is resolved relative to
the module both during graph building and at runtime.

Fixes #15559

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -181,3 +181,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.base64-simd]
 opt-level = 3
+
+[patch.crates-io]
+deno_graph = { path = "../deno_graph" }
```
Comment (kitsonk) [inline Cargo.toml:186]: oooh, TIL...

**Review by kitsonk — APPROVED** [review]
LGTM, pending upstream package publish.

Clean and effective!

File: Cargo.toml (hunk 1 above)
Comment (dsherret) [reply Cargo.toml:186]: Me too. This is much cleaner than updating everything directly. I'll have to switch my "to local source" scripts to use this.

Comment (bartlomieju) [reply Cargo.toml:186]: It's nice, but limited. Yesterday I needed to override both `deno_graph` and `deno_ast` (which `deno_graph` depends on) and wasn't able to do that

Comment (dsherret) [reply Cargo.toml:186]: Oh, that sucks then 😂

**Pull request merged**